app/thirdparty/java/other/writeily/model/WrMarkorSingleton.java

```diff
@@ -66,60 +67,66 @@ private boolean saneMove(final File file, final File dest) {
         return saneCopy(file, dest) && !file.equals(dest);
     }
 
-    public boolean moveFile(final File file, final File dest, final Context context) {
+    public boolean moveFile(final File file, final File dest, final Context context, AppSettings appSettings) {
         if (saneMove(file, dest) && !dest.exists()) {
-            boolean renameSuccess;
-            try {
-                renameSuccess = file.renameTo(dest);
-            } catch (Exception e) {
-                renameSuccess = false;
-            }
-            return (renameSuccess || (copyFile(file, dest) && deleteFile(file, context)));
+            return (copyFile(file, dest, appSettings) && deleteFile(file, context, appSettings));
         }
         return false;
     }
 
-    public boolean copyFile(final File file, final File dest) {
+    public boolean copyFile(final File file, final File dest, AppSettings appSettings) {
         if (saneCopy(file, dest) && !dest.exists()) {
             if (file.isDirectory()) {
                 if (dest.mkdir()) {
                     boolean success = true;
+                    appSettings.copyFavouriteFile(file, dest);
                     for (final File dirFile : file.listFiles()) {
                         // Merge not supported, dest here will always be available
-                        success &= this.copyFile(dirFile, new File(dest, dirFile.getName()));
+                        success &= this.copyFile(dirFile, new File(dest, dirFile.getName()), appSettings);
                     }
                     return success;
                 }
                 return false;
             } else {
                 FileUtils.copyFile(file, dest);
+                appSettings.copyFavouriteFile(file, dest);
                 return true;
             }
         }
         return false;
     }
 
-    public boolean deleteFile(final File file, final Context context) {
+    public boolean deleteFile(final File file, final Context context, AppSettings appSettings) {
         if (file.isDirectory()) {
             for (final File childFile : file.listFiles()) {
-                deleteFile(childFile, context);
+                deleteFile(childFile, context, appSettings);
```

Comment:
I would suggest not coupling move and delete to an instance of appSettings. These are more like general utility functions. I think we should instead call delete -> check success -> remove from favorites from the appropriate call sites.

Reply:
Yes, I also think that coupling it here is not such a good choice. I'll do the updates.

Reply:
as said earlier: you don't need to remove something from favorites at all. we don't need that explicit removal. The getter for favorites automatically does (& has to) filter out all non existing favorites. So a new call to the getter is all you should need. And that once.

Reply:
That's the previous design, and it cause a behavior which seems like a bug to me: favorite a file, delete it, and create that same file again, the file would show up as favorite. Is that a desired behavior?

Reply:
then there is a issue in that function. it should drop that favorite.

though we also have these things all over the place. all per-file settings are tied to specific files, and they are not resetted/dropped upon file removal. so not sure if generally the SharedPreference way with filename as key is the holy grail ^^. (speaking about things like atuo-format, font-size, ..)